split.py

```diff
@@ -1,43 +1,110 @@
 import bpy
 import bl_operators
 import bmesh
+import math
 from math import ceil
+import numpy as np
 
 obj = bpy.context.active_object
 
-#colours
+# colours
 mat_zero = bpy.data.materials.new("mat_zero")
 mat_zero.diffuse_color = (0, 0, 0, 1)
 mat_one = bpy.data.materials.new("mat_one")
 mat_one.diffuse_color = (1, 0, 0, 1)
+mat_two = bpy.data.materials.new("mat_two")
+mat_two.diffuse_color = (0, 1, 0, 1)
+mat_three = bpy.data.materials.new("mat_three")
+mat_three.diffuse_color = (0, 0, 1, 1)
 
 obj.data.materials.append(mat_zero)
 obj.data.materials.append(mat_one)
+obj.data.materials.append(mat_two)
+obj.data.materials.append(mat_three)
 
 me = obj.data
 bm = bmesh.new()
 bm.from_mesh(me)
 
 
-#mark seam so that kerfed part stays in one piece
+def angle_between_normals(a, b):
+    dp = np.dot(a, b)
+    if (dp > 1):
+        dp = 1
+    return math.degrees(math.acos(dp))
+
+def remove_duplicates(duplicate):
```

Comment:
The reason that there are duplicates was that I was adding edges to a list going face by face, and because edges are sharing by two faces, they would get added twice. There is probably a much cleaner way to do this, I'll look into it